src/par_iter/internal.rs

```diff
@@ -7,14 +7,20 @@ use join;
 use super::IndexedParallelIterator;
 use super::len::*;
 
+pub trait ProducerContinuation<ITEM> {
+    type Output;
+    fn with_producer<'p, P>(self, producer: P, shared: &'p P::Shared) -> Self::Output
+        where P: Producer<'p, Item=ITEM>;
+}
+
```

Comment:
So this trait and `Producer` are in the `internal` module - does that mean that clients would not need to impl their own `Producer`s if they wanted to create custom parallel iterators?